planning_interface/move_group_interface/src/move_group.cpp

```diff
@@ -689,6 +689,38 @@ class MoveGroup::MoveGroupImpl
     }
   }
 
+  bool validatePlan(const Plan &plan)
+  {
+    robot_state::RobotStatePtr current_state;
+    if (!getCurrentState(current_state))
+      return false;
+    if (plan.trajectory_.joint_trajectory.points.empty())
+      return true;
+
+    const trajectory_msgs::JointTrajectory &trajectory = plan.trajectory_.joint_trajectory;
+    const std::vector<double> &positions = trajectory.points.front().positions;
+    size_t n = trajectory.joint_names.size();
+    if (positions.size() != n)
+      return false;
+
+    for (size_t i = 0; i < n; ++i)
+    {
+      const robot_model::JointModel *jm = robot_model_->getJointModel(trajectory.joint_names[i]);
+      if (!jm)
+      {
+        ROS_ERROR("unknown joint in trajectory: %s", trajectory.joint_names[i]);
+        return false;
+      }
+      // TODO: check multi-DoF joints
+      if (fabs(current_state->getJointPositions(jm)[0] - positions[i]) < FLT_EPSILON)
```

Comment:
moveit uses std::numeric_limits<double>::epsilon() thought, and not the C equivalent